src/app/services/git.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, Inject, Provider } from '@angular/core';
+import { ConfigService } from './config.service';
+import * as os from 'os';
+import * as fs from 'fs';
+import * as path from 'path';
+
+@Injectable()
+export class GITService {
+  currentBranch: string;
+  cwd: string;
+
+  constructor(@Inject(ConfigService) private _config: ConfigService) {
+    this.currentBranch = null;
+    this.cwd = null;
+  }
+
+  hasGit(): boolean {
+    if (!this.cwd) { return false; }
+    let gitDir: string = path.join(this.cwd, '.git');
+
+    if (gitDir.startsWith('~')) { gitDir = gitDir.replace(/~/, os.homedir()); }
+
+    if (fs.existsSync(gitDir)) {
+      try {
+        let head: string = fs.readFileSync(path.join(gitDir, 'HEAD')).toString();
+        let parts = head.split('/') || null;
+        this.branch = parts[parts.length - 1] || null;
```

Comment:
this now always work unfortunately, for example `pr/81` you get only `81`.
what would be the solution for all cases 
```ts
child_process.exec(`git rev-parse --abbrev-ref HEAD`, (err, stderr, stdour) => {
  if (!err) {
   this.branch = stdout;
  }
});
```